frontend/app/services/personal-information-service.server.ts

```diff
@@ -185,6 +189,22 @@ function createPersonalInformationService() {
       },
     };
   }
+  async function getPersonalInformationIntoSession(session: Session, request: Request, sin?: string) {
+    if (session.has('personalInformation')) {
+      log.debug(`User has personal information object;`);
+      return session.get('personalInformation');
+    }
+    if (!sin) {
+      throw new HttpResponse('SIN must be present', { status: 401 });
+    }
 
-  return { getPersonalInformation };
+    const personalInformation = await getPersonalInformation(sin);
+    if (!personalInformation) {
+      log.debug(`No personal information found for SIN`);
+      throw redirectWithLocale(request, '/data-unavailable');
```

Comment:
Consider allowing the consumer specify the redirect URL if no personal information is found. That way, this service function is decoupled from the application flow logic.